lib/utils/log/eventlog/eventlog_windows.go

```diff
@@ -0,0 +1,144 @@
+func Install(logName, source, msgFile string, useExpandKeyForMsgFile bool) error {
+	return trace.Wrap(
+		install(logName, source, msgFile, useExpandKeyForMsgFile, eventlog.Info|eventlog.Warning|eventlog.Error),
+	)
+}
+
+// Remove removes the registry entry for source in logName. Requires admin privileges to run. It
+// does not remove logs generated thus far, but Event Log will no longer be able to properly
+// interpolate them – event data sent with log messages can still be manually extracted from them
+// though using Get-WinEvent.
+// It also does not remove the registry key for the custom log.
+func Remove(logName, source string) error {
+	return trace.Wrap(remove(logName, source))
+}
+
+func makeLogKeyName(logName string) string {
+	return `SYSTEM\CurrentControlSet\Services\EventLog\` + logName
+}
+
+// install is a copy of [eventlog.Install] that makes it possible to create a new log under logName
+// instead of adding a new event source under the Application log. Variable names are kept the same
+// for the most part so that it's easy to compare this copy to the original after package updates.
+//
+// install modifies PC registry to allow logging with an event source src.
+// It adds all required keys and values to the event log registry key.
+// Install uses msgFile as the event message file. If useExpandKey is true,
+// the event message file is installed as REG_EXPAND_SZ value,
+// otherwise as REG_SZ. Use bitwise of log.Error, log.Warning and
+// log.Info to specify events supported by the new event source.
+func install(logName, source, msgFile string, useExpandKey bool, eventsSupported uint32) error {
+	addKeyName := makeLogKeyName(logName)
+	// registry.WRITE is needed for setting values on appKey and creating a sub key (sk).
+	// registry.QUERY_VALUE is needed for reading values from appKey (MaxSize).
+	// https://learn.microsoft.com/en-us/windows/win32/sysinfo/registry-key-security-and-access-rights
+	appkey, _, err := registry.CreateKey(registry.LOCAL_MACHINE, addKeyName, registry.WRITE|registry.QUERY_VALUE)
+	if err != nil {
+		return trace.Wrap(err, "creating registry key for log")
+	}
+	defer appkey.Close()
+
+	// Set MaxSize for custom log if not already set. Without this, the default size for a custom log
+	// is just 1 MB.
+	//
+	// Docs for registry values under an Eventlog key such as MaxSize:
+	// https://learn.microsoft.com/en-us/windows/win32/eventlog/eventlog-key
+	if _, _, err = appkey.GetIntegerValue("MaxSize"); err != nil {
+		if !errors.Is(err, registry.ErrNotExist) {
+			return trace.Wrap(err, "checking max size of log")
+		}
+
+		const defaultLogMaxSizeBytes = 20 * 1024 * 1024 // 20 MB
+		if err := appkey.SetDWordValue("MaxSize", defaultLogMaxSizeBytes); err != nil {
+			return trace.Wrap(err, "setting max size for log")
+		}
+	}
+
+	sk, alreadyExist, err := registry.CreateKey(appkey, source, registry.SET_VALUE)
+	if err != nil {
+		return trace.Wrap(err)
```

Comment:
I like to add a custom message to the `trace.Wrap` whenever calling an external library to make it clearer what we were trying to do

Reply:
This code is mostly copied from x/sys, hence why it was missing messages in `trace.Wrap`, but I added them now.